cmd/src/batch_remote.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/cockroachdb/errors"
+	"github.com/sourcegraph/sourcegraph/lib/output"
+	"github.com/sourcegraph/src-cli/internal/batches/service"
+	"github.com/sourcegraph/src-cli/internal/batches/ui"
+)
+
+func init() {
+	usage := `'src batch remote' runs a batch spec on the Sourcegraph instance.
+
+Usage:
+
+    src batch remote [-f FILE]
+    src batch remote FILE
+
+Examples:
+
+    $ src batch remote -f batch.spec.yaml
+
+`
+
+	flagSet := flag.NewFlagSet("remote", flag.ExitOnError)
+	flags := newBatchExecutionFlags(flagSet)
+
+	var (
+		fileFlag = flagSet.String("f", "", "The name of the batch spec file to run.")
+	)
+
+	handler := func(args []string) error {
+		// Various bits of Batch Changes boilerplate.
+		ctx := context.Background()
+
+		if err := flagSet.Parse(args); err != nil {
+			return err
+		}
+
+		file, err := getBatchSpecFile(flagSet, fileFlag)
+		if err != nil {
+			return err
+		}
+
+		svc := service.New(&service.Opts{
+			Client: cfg.apiClient(flags.api, flagSet.Output()),
+		})
+
+		if err := svc.DetermineFeatureFlags(ctx); err != nil {
+			return err
+		}
+
+		out := output.NewOutput(flagSet.Output(), output.OutputOpts{Verbose: *verbose})
+		ui := &ui.TUI{Out: out}
+
+		// OK, now for the real stuff. We have to load in the batch spec, and we
+		// may as well validate it at the same time so we don't even have to go to
+		// the backend if it's invalid.
+		ui.ParsingBatchSpec()
+		_, raw, err := parseBatchSpec(file, svc)
+		if err != nil {
+			ui.ParsingBatchSpecFailure(err)
+			return err
+		}
+		ui.ParsingBatchSpecSuccess()
+
+		// We're going to need the namespace ID, so let's figure that out.
+		ui.ResolvingNamespace()
+		namespaceID, err := svc.ResolveNamespace(ctx, flags.namespace)
+		if err != nil {
+			return err
+		}
+		ui.ResolvingNamespaceSuccess(namespaceID)
+
+		ui.SendingBatchSpec()
+		batchSpecID, err := svc.UpsertBatchSpecInput(
+			ctx,
+			raw,
+			namespaceID,
+			flags.allowIgnored,
+			flags.allowUnsupported,
+			flags.clearCache,
+		)
+		if err != nil {
+			return err
+		}
+		ui.SendingBatchSpecSuccess()
+
+		// Wait for the workspaces to be resolved.
+		ui.ResolvingWorkspaces()
+		ticker := time.NewTicker(1 * time.Second)
+		defer ticker.Stop()
+		for range ticker.C {
+			res, err := svc.GetBatchSpecWorkspaceResolution(ctx, batchSpecID)
+			if err != nil {
+				return err
+			}
+
+			if res.State == "FAILED" {
+				return errors.Newf("workspace resolution failed: %s", res.FailureMessage)
+			} else if res.State == "COMPLETED" {
+				break
+			}
+		}
+		ui.ResolvingWorkspacesSuccess()
+
+		// We have to enqueue this for execution with a separate operation.
+		//
+		// TODO: when the execute flag is wired up in the upsert mutation, just set
+		// it there and remove this.
```

Comment:
is this still a todo?

Reply:
[Yep.](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@5a1824e22366d4e7ff5e4f0776509993bf1a9e3a/-/blob/cmd/frontend/graphqlbackend/batches.graphql?L1405:9)

Reply:
Oh, I think I misunderstood the comment. Makes sense!